### PR TITLE
[#9480] Add unit tests for feedback-path.pipe.spec.ts and question-edit-form-component.spec.ts

### LINKED ISSUE #9480: Add front-end tests for all Angular components in v7
To add test cases (testing the business logic with usual testing and HTML generation with snapshot testing) on all frontend `spec.ts` files

### PR DESCRIPTION
**Pull request by daongochieu2810**
Part of #9480

Add more unit tests for feedback-path.pipe.spec.ts and question-edit-form-component.spec.ts

feedback-path.pipe.spec.ts: tested 'transform' function of ```GiverTypeDescriptionPipe``` and ```RecipientTypeDescriptionPipe```

question-edit-form-component.spec.ts: tested 'formModel' and 'triggerModelChange' functions

### CODE REVIEW

**Review by ccyccyccy — APPROVED** [review]
LGTM!

File: src/web/app/components/question-edit-form/feedback-path.pipe.spec.ts [hunk 1]
```diff
@@ -1,15 +1,90 @@
...
+  });
+
+  it('transform FeedbackParticipantType.TEAMS', () => {
+    expect(pipe.transform(FeedbackParticipantType.TEAMS))
+      .toBe('Teams in this course');
+  });
+
+  it('transform default', () => {
+    expect(pipe.transform('' as FeedbackParticipantType))
+      .toBe('Unknown');
```
Comment (ChooJeremy) [inline src/web/app/components/question-edit-form/feedback-path.pipe.spec.ts:33]: Let's test an existing (but invalid) type too and make sure it works.
For example, `FeedbackParticipantType.OWN_TEAM` should return `Unknown` instead of `Giver's team`

File: src/web/app/components/question-edit-form/feedback-path.pipe.spec.ts [hunk 2]
```diff
@@ -1,15 +1,90 @@
...
+  });
+
+  it('transform FeedbackParticipantType.NONE', () => {
+    expect(pipe.transform(FeedbackParticipantType.NONE))
+      .toBe('Nobody specific (For general class feedback)');
+  });
+
+  it('transform default', () => {
+    expect(pipe.transform('' as FeedbackParticipantType))
+      .toBe('Unknown');
```
Comment (ChooJeremy) [inline src/web/app/components/question-edit-form/feedback-path.pipe.spec.ts:87]: Same here

**Review by ChooJeremy — CHANGES_REQUESTED** [review]
Some minor changes

File: angular.json [hunk 3]
```diff
@@ -115,5 +115,8 @@
       }
     }
   },
-  "defaultProject": "teammates"
-}
+  "defaultProject": "teammates",
+  "cli": {
+    "analytics": "5b44a57d-2fa6-4bda-8654-8d8d27cc0df1"
```
Comment (ChooJeremy) [inline angular.json:120]: Where is this from/why is this needed?

Comment (daongochieu2810) [reply angular.json:120]: Must have added it by mistake, I will remove it

**Review by ChooJeremy — APPROVED** [review]
